src/webgpu/shader/validation/expression/call/builtin/asinh.spec.ts

```diff
@@ -0,0 +1,72 @@
+const builtin = 'asinh';
+export const description = `
+Validation tests for the ${builtin}() builtin.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import {
+  TypeF16,
+  TypeF32,
+  elementType,
+  kAllFloatScalarsAndVectors,
+  kAllIntegerScalarsAndVectors,
+} from '../../../../../util/conversion.js';
+import { isRepresentable } from '../../../../../util/floating_point.js';
+import { linearRange } from '../../../../../util/math.js';
+import { ShaderValidationTest } from '../../../shader_validation_test.js';
+
+import {
+  kConstantAndOverrideStages,
+  stageSupportsType,
+  validateConstOrOverrideBuiltinEval,
+} from './const_override_validation.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+g.test('values')
+  .desc(
+    `
+Validates that constant evaluation and override evaluation of ${builtin}() rejects invalid values
+`
+  )
+  .params(u =>
+    u
+      .combine('stage', kConstantAndOverrideStages)
+      .combine('type', kAllFloatScalarsAndVectors)
+      .filter(u => stageSupportsType(u.stage, u.type))
+      .combine('value', linearRange(-2000, 2000, 10))
```

Comment:
I thought about using fullRangeForType.
but asinh is so well behaved it's not likely to be worth it.

*Might* be worth saying why this doesn't use fullrange.

Reply:
Why not both? Combined it with `fullRangeForType`.